### PR TITLE
Added windowsAdminPassword to the parameters than can be a reference KeyVault

### PR DESCRIPTION
**Pull request by askew**
A number of ACS template parameters can be set as KeyVault references; the certificates and Service Principal key, however the Windows administrator password cannot.

This change adds `windowsAdminPassword` to the KeyVault secret parameters.

#354

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/355)

### CODE REVIEW

**Comment by askew** [comment]
I've modified the testdata in key-vault-params/kubernetes.json by changing the second agent pool to type Windows and including the KeyVault reference for the password.

I've left the password as plain-text in the testdata/windows tests so that both cases are being tested.

**Comment by rjtsdl** [comment]
@anhowe @JiangtianLi, how do you think of we keep testdata/windows folder's windowsPassword as plain-text?

**Comment by rjtsdl** [comment]
I have approved this. Need resolve conflict. And make sure get approve from @JiangtianLi  or @anhowe . Thanks

**Comment by JiangtianLi** [comment]
Chatted with Jingtao offline. If we don't have plain-text windowsAdminPassword test covered elsewhere, we may leave windows folder as is. Anthony, what do you think? We can rearrange tests as necessary.

File: pkg/acsengine/testdata/key-vault-params/kubernetes_expected.json [hunk 1]
```diff
@@ -675,9 +675,25 @@
         "description": "The azure deploy environment. Currently support: AzureCloud, AzureChinaCloud"
       },
       "type": "string"
+    },
+    "windowsAdminPassword": {
+      "metadata": {
+        "description": "Password for the Windows Swarm Agent Virtual Machines."
```
Comment (JiangtianLi) [inline pkg/acsengine/testdata/key-vault-params/kubernetes_expected.json]: Not related to this commit, this is a typo in windowsparams.t and we can fix in another issue/PR.

**Comment by anhowe** [comment]
@acs-bot test this please

**Comment by anhowe** [comment]
@askew please resolve merge conflicts and we will merge.  This fixes #354

**Comment by JackQuincy** [comment]
@askew said in email he doesn't have time to finish this. Since it was small I decided to pick it up and sent a new PR #567

**Pull request closed without merging**